fdbserver/ptxn/TLogPeekCursor.actor.h

```diff
@@ -148,20 +240,25 @@ class StorageTeamPeekCursor : public PeekCursorBase {
 	// pArena_ is used to store the serialized data for further use, e.g. making MutationRefs still available after the
 	// cursor is destroyed. If pArena_ is nullptr, any reference to the peeked data will be invalidated after the cursor
 	// is destructed.
+	// reportEmptyVersion_ flag will allow an empty version trigger an EmptyMessage, if set true
 	StorageTeamPeekCursor(const Version& version_,
 	                      const StorageTeamID& storageTeamID_,
 	                      TLogInterfaceBase* pTLogInterface_,
-	                      Arena* pArena_ = nullptr);
+	                      Arena* pArena_ = nullptr,
+	                      const bool reportEmptyVersion_ = false);
 
 	StorageTeamPeekCursor(const Version& version_,
 	                      const StorageTeamID& storageTeamID_,
 	                      const std::vector<TLogInterfaceBase*>& pTLogInterfaces_,
-	                      Arena* arena_ = nullptr);
+	                      Arena* pArena_ = nullptr,
+	                      const bool reportEmptyVersion_ = false);
+
+	bool isEmptyVersionsIgnored() const { return !reportEmptyVersion; }
 
 	const StorageTeamID& getStorageTeamID() const;
 
-	// Returns the last version being pulled
-	const Version& getLastVersion() const;
+	// Returns the starting verion for the cursor. The cursor will start from the begin version.
```

Comment:
is "starting version" the same as "begin version"? In the code base, we typically use "begin/end versions", not "starting/last versions". It's better to stick with conventions.